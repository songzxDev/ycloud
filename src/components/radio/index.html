<div id="y-radio" class="y-radio" data-bind="foreach: {data: dataList, as: 'item'}, css:{'y-radio-hor':direction == 'horizontal','y-radio-ver':direction == 'vertical'}">
  <y-radio-item params="label: item.label, value: item.value, 
  parent: $parent.dataList "></y-radio-item>
</div>
