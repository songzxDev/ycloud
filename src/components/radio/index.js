import template from './index.html'
import ko from 'knockout'
import item from './radioitem'
const PREFIX = 'y-'
ko.components.register(PREFIX + item.name, {
  viewModel: item.init,
  template: item.template
})
function init (params) {
  if (params.dataList && params.dataList.subscribe) {
    this.dataList = params.dataList
  } else {
    this.dataList = ko.observableArray(params.dataList || [])
  }
  this.value = params.value || ko.observableArray([])
  this.direction = params.direction || 'horizontal'
}

export default {
  name: 'radio',
  init,
  template
}
