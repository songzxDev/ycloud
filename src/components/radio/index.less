.y-radio-ver .y-radio-item{
  display: block;
}
.y-radio-item{
  cursor: pointer;
  font-size: @f-size-base;
  display: inline-block;
  margin-right: 25px;
  input{
    width: 100%;
    height: 100%;
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    z-index: 1;
    cursor: pointer;
    opacity: 0;
  }
  .y-radio-label{
    position: relative;
    top: 1px;
  }
}
.y-radio-item--inner-warpper{
  display: inline-block;
  vertical-align: middle;
  white-space: nowrap;
  cursor: pointer;
  outline: 0;
  line-height: 1;
  position: relative;
}
.y-radio-item-inner{
  display: inline-block;
  width: 14px;
  height: 14px;
  position: relative;
  top: 2px;
  left: 0;
  border: 1px solid #dddee1;
  border-radius: 50%;
  background-color: #fff;
  transition: border-color .2s ease-in-out,background-color .2s ease-in-out;
}
.y-radio-item-checked .y-radio-item-inner{
  border-color: @primary-color;
  &:after{
    content: "";
    display: table;
    width: 8px;
    height: 8px;
    position: absolute;
    top: 2px;
    left: 2px;
    border-radius: 6px;
    border-top: 0;
    border-left: 0;
    background-color: #2d8cf0;
    opacity:1;
    transform: rotate(45deg) scale(1);
    transition: all .2s ease-in-out;
  }
}
.y-radio-item-disabled{
  color: @f-color-light;
  cursor: not-allowed;
  .y-radio-item-inner{
    background-color: #eee;
  }
}
