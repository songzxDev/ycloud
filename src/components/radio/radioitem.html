<label class="y-radio-item" data-bind="css:{'y-radio-item-checked': checked}">
  <span class="y-radio-item--inner-warpper">
    <span class="y-radio-item-inner">
      <input type="radio" data-bind="attr:{name:radioname},checked: checked, event:{change:change }">
    </span>
  </span>
  <span class="y-radio-label" data-bind="text:value.label"></span>
</label>
