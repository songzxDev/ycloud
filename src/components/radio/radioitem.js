import template from './radioitem.html'
import ko from 'knockout'
function init (params) {
  this.value = params.value
  this.currentValue = params.currentValue
  this.radioname = params.radioname
  this.checked = ko.computed(() => {
    // todo 默认是value/label 后续要改成其他自定义格式
    return this.currentValue() === this.value.value
  })
  this.change = (data) => {
    this.currentValue(this.value)
  }
}
export default {
  name: 'radio-item',
  init,
  template
}
