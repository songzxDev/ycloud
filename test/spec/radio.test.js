import ko from 'knockout'
import radio from '../../src/components/radio/index.js'
import $ from 'jquery'
describe('y-radio组件', () => {
  const PREFIX = 'y-'
  ko.components.register(PREFIX + radio.name, {
    viewModel: radio.init,
    template: radio.template,
    synchronous: true
  })
  ko.cleanNode(document.getElementsByTagName('body')[0])
  let vm = {
    title: ko.observable('test'),
    radioDataList: ko.observableArray([{value: 1,label:'男'}, {value: 0,label:'女'}]),
    radioValue: ko.observable(1)
  }
  document.body.innerHTML = '<y-radio params="dataList:radioDataList,value:radioValue"></y-radio>'
  ko.applyBindings(vm)
  var el = $('body')
  it("组件注册测试", () => {
    expect(radio).not.toBeNull();
    expect(ko.components.isRegistered(PREFIX + radio.name)).toBe(true);
  })
  it('生成radio测试', () => {
    expect($('body').find('.y-radio').length > 0).toEqual(true)
  })
  it('hor默认属性设置', () => {
    expect($('body').find('.y-radio-hor').length > 0).toEqual(true)
  })
  // todo radio方向测试
  it('快照', () => {
    expect($('body').get(0)).toMatchSnapshot()
  })
})
