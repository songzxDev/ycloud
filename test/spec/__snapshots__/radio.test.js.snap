// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`y-radio组件 快照 1`] = `
<body>
  <y-radio
    params="dataList:radioDataList,value:radioValue"
  >
    <div
      class="y-radio y-radio-hor"
      data-bind="foreach: {data: dataList, as: 'item'}, css:{'y-radio-hor':direction == 'horizontal','y-radio-ver':direction == 'vertical'}"
      id="y-radio"
    >
      
  
      <y-radio-item
        params="radioname: $parent.radioname, value: item, currentValue: $parent.value"
      />
      

      
  
      <y-radio-item
        params="radioname: $parent.radioname, value: item, currentValue: $parent.value"
      />
      

    </div>
    

  </y-radio>
</body>
`;
